File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/busobjs/VB/BusObj.vb
Imports System
Imports System.Data
Imports System.Data.SQL
Imports DataLayer

Namespace BusinessLayer 

  Public Class BusObjVB

    Private Data As DataObjVB

    Public Sub New() 

      MyBase.New()
      Data = New DataObjVB("server=localhost;uid=sa;pwd=;database=grocertogo")
    End Sub

    Public Function GetCategories() As DataView

       Return Data.GetCategories()
    End Function

    Public Function GetProductsForCategory(Category As String, CustomerId As Integer) As DataView

       Dim View As DataView
       View = Data.GetProductsForCategory(Category)

       Dim Discount As Double = 0
       If (CustomerId >= 25) And (CustomerId < 50)
         Discount = .50
       Else If (CustomerId >= 50) And (CustomerId < 75)
         Discount = 1.00 
       Else If (CustomerId >= 75) And (CustomerId < 100)
         Discount = 1.50
       End If

       Dim I As Integer
       For I = 0 To View.Count - 1
         View(I)("UnitPrice") = CDbl(View(I)("UnitPrice").ToString()) - Discount
       Next

       Return View
    End Function

  End Class

End Namespace

File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/busobjs/BusObj.cs

namespace BusinessLayer 
{
using System;
using System.Data;
using System.Data.SQL;
using DataLayer;

  public class BusObj 
  {
    private DataObj data;

    public BusObj() 
    {
      data = new DataObj("server=localhost;uid=sa;pwd=;database=grocertogo");
    }

    public DataView GetCategories() 
    {
       return data.GetCategories();
    }

    public DataView GetProductsForCategory(String category, int customerid) 
    {
       DataView view = data.GetProductsForCategory(category);

       double discount = 0;
       if ((customerid >= 25)&&(customerid < 50))
       {
         discount = .50; 
       }
       else if ((customerid >= 50)&&(customerid < 75))
       {
         discount = 1.00; 
       }
       else if ((customerid >= 75)&&(customerid < 100))
       {
         discount = 1.50; 
       }

       for (int i=0; i<view.Count; i++)
       {
         view[i]["UnitPrice"] = Double.Parse(view[i]["UnitPrice"].ToString()) - discount;
       }

       return view;
    }
  }

}